File: 1-patrons-de-creations/abstract-factory/code-typescript/factories/concreteFactories/macFactory.ts
import { Button } from "../../guiElements/interfaces/button.js";
import { Checkbox } from "../../guiElements/interfaces/checkbox.js";
import { MacButton } from "../../guiElements/macElements/macButton.js";
import { MacCheckbox } from "../../guiElements/macElements/macCheckbox.js";
import { GUIFactory } from "../guiFactory.js";

export class MacFactory implements GUIFactory {
  createButton(): Button {
    return new MacButton();
  }

  createCheckbox(): Checkbox {
    return new MacCheckbox();
  }
}

File: 1-patrons-de-creations/abstract-factory/code-java/factories/concrete_factories/MacFactory.java
package factories.concrete_factories;

import factories.abstract_factories.GuiFactory;

import gui_elements.interfaces.*;

import gui_elements.mac_elements.*;

public class MacFactory implements GuiFactory {

    public void MacFactory() {};

    //create button
    public Button createButton() {
        return new MacButton();
    }

    //create checkbox
    public Checkbox createCheckbox() {
        return new MacCkeckbox();
    }
}